### ISSUE TITLE
press Ctrl+shift+f10 won't run main()

### ISSUE THREAD

**Issue by Sherlock-Holo** [intellij-rust/intellij-rust#6627]
## Environment

* **IntelliJ Rust plugin version:** 0.3.139.3589-203-nightly
* **Rust toolchain version:** rustc 1.51.0-nightly (da305a2b0 2021-01-05)
* **IDE name and version:** clion 2020.3.1
* **Operating system:** windows 10

## Problem description
```rust
#![feature(type_alias_impl_trait)]
#![feature(generic_associated_types)]

use async_trait_static::async_trait;
use serde::Deserialize;

#[derive(Deserialize, Debug)]
struct T {
    a: u32,
    b: u32,
}

fn main() {
    let data = br#"{"a": 1, "b": }"#;

    let t = serde_json::from_reader::<_, T>(&data[..]).unwrap();

    println!("{:?}", t);
}
```

in the main.rs page, press Ctrl+shift+f10, it open the terminal, output is
```
C:\Users\tttt\Documents\git\rust-learn\src>C:/Users/tttt/Documents/git/rust-learn/src/main.rs

C:\Users\tttt\Documents\git\rust-learn\src>
```

and my openning vscode is open and show this file content: main.rs

it should use `cargo run` to run the codes, not open my vscode

## Steps to reproduce

press ctrl+shift+f10 in the main.rs page

**Comment by Sherlock-Holo** [comment]
I found a min codes to reproduce the bug
```rust
// #![feature(type_alias_impl_trait)]               decommonet it will cause this bug
#![feature(generic_associated_types)]

use async_trait_static::async_trait;
use futures::StreamExt;
use serde::Deserialize;
use serde_json::Value;
use tokio_serde::{Framed, SymmetricallyFramed};
use tokio_serde::formats::*;
use tokio_util::codec::{FramedRead, LengthDelimitedCodec};

#[tokio::main]
async fn main() {
    let data = br#"{"a": 1, "b": }"#;

    let length_delimited = FramedRead::<&[u8], _>::new(&data[..], LengthDelimitedCodec::new());

    let mut framed = SymmetricallyFramed::new(length_delimited, SymmetricalJson::<Value>::default());

    let value = framed.next().await.unwrap().unwrap();

    println!("{}", value);
}

#[derive(Deserialize, Debug)]
struct T {
    a: u32,
    b: u32,
}
```

**Comment by Sherlock-Holo** [comment]
it seems if contain `#![feature(type_alias_impl_trait)]` will cause this problem
I try this feature gate in another real using project too, problem same

**Comment by Undin** [comment]
I reproduced this bug. But a concrete feature doesn't matter. Even concrete attribute doesn't matter.
It can be reproduced when the file starts with `#!`. It's a bug in shell plugin in your IDE
I've created the corresponding issue - https://youtrack.jetbrains.com/issue/IDEA-259116

As a workaround, I may suggest just add any other symbol before `#!`

**Comment by Undin** [comment]
https://youtrack.jetbrains.com/issue/IDEA-259116 is fixed in the platform. The fix will be available in the first EAP of 2021.1
